src/lib/y2storage/proposal/autoinst_devices_creator.rb

```diff
@@ -86,6 +88,66 @@ def best_distribution(planned_partitions, disk_names)
 
         calculator.best_distribution(planned_partitions, spaces)
       end
+
+    private
+
+      # Creates planned partitions in the given devicegraph
+      #
+      # @param planned_partitions [Array<Planned::Partition>] Devices to create/reuse
+      # @param disk_names [Array<String>] Disks to consider
+      # @return [PartitionCreatorResult]
+      def create_partitions(planned_partitions, disk_names)
+        log.info "Partitions to create: #{planned_partitions}"
+
+        dist = best_distribution(planned_partitions, disk_names)
+        raise Error, "Could not find a valid partitioning distribution" if dist.nil?
+        part_creator = Proposal::PartitionCreator.new(original_graph)
+        part_creator.create_partitions(dist)
+      end
+
+      # Creates volume groups in the given devicegraph
+      #
+      # @param vgs         [Array<Planned::LvmVg>]           List of planned volume groups to add
+      # @param devicegraph [Devicegraph]                     Starting point
+      # @param devices_map [Hash<String,Planned::Partition>] Map of device names and partitions
+      # @return [Devicegraph] Devicegraph containing the specified volume groups
+      def create_volume_groups(vgs, devicegraph, devices_map)
+        vgs.reduce(devicegraph) do |graph, vg|
+          lvm_creator = Proposal::LvmCreator.new(graph)
+          pvs = find_pvs_for(devices_map, vg.volume_group_name)
+          lvm_creator.create_volumes(vg, pvs)
+        end
+      end
+
+      # Finds physical volumes for a volume group
+      #
+      # @param devices_map [Hash<String,Planned::Partition>] Map of device names and partitions
+      # @param vg_name     [String] Volume group name
+      # @return [Array<String>] Physical volumes device names (eg. ["/dev/sda1"])
+      #
+      # @see Y2Storage::Proposal::PartitionCreator#create_partitions
+      def find_pvs_for(devices_map, vg_name)
+        devices_map.select { |_k, v| v.lvm_volume_group_name == vg_name }.keys
+      end
+
+      # Reuses partitions for the given devicegraph
+      #
+      # @param reused_partitions [Array<Planned::Partition>] Partitions to reuse
+      # @param devicegraph       [Devicegraph]               Devicegraph to reuse partitions
+      def reuse_partitions(reused_partitions, devicegraph)
+        reused_partitions.each { |r| r.reuse!(devicegraph) }
+      end
+
+      # Reuses volume groups for the given devicegraph
+      #
+      # @param reused_vgs  [Array<Planned::LvmVg>] Volume groups to reuse
+      # @param devicegraph [Devicegraph]           Devicegraph to reuse partitions
+      def reuse_vgs(reused_vgs, devicegraph)
+        reused_vgs.each do |vg|
+          vg.reuse!(devicegraph)
+          vg.lvs.each { |v| v.reuse!(devicegraph) }
```

Comment:
Maybe a comment here clarifying that the fact that we call `#reuse!` for all the planned LVs doesn't imply all of them are going to be reused (that's decided by each PlannedDevice). That's actually fine, but in the case of partitions we only call `#reuse!` for those for which `#reuse?` is true. That can be confusing. I don't care if we fix it with a comment or with a `if v.reuse?`.